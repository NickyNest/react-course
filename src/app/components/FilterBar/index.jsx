import React from 'react';
import {func, string} from 'prop-types';
import {Button} from 'semantic-ui-react';
import * as actions from 'utils/actions';

const FilterBar = ({changeShowMode, removeCompleted, changeSortMode, currentMode}) => {
    const onClick = showMode => {
        changeShowMode(showMode);
    };

    const onChageSort = sortMode => {
        changeSortMode(sortMode);
    };

    return (
        <div>
            <Button onClick={() => onClick(actions.SHOW_ALL)}>All</Button>
            <Button onClick={() => onClick(actions.SHOW_COMPLETED)}>Completed</Button>
            <Button onClick={() => onClick(actions.SHOW_PENDING)}>Pending</Button>
            <Button onClick={removeCompleted} disabled={currentMode === actions.SHOW_PENDING}>Remove completed</Button>
            <Button onClick={() => onChageSort(actions.SORT_CREATED_DATE_UP)}>Date UP</Button>
            <Button onClick={() => onChageSort(actions.SORT_CREATED_DATE_DOWN)}>Date DOWN</Button>
        </div>
    );
};

FilterBar.propTypes = {
    changeShowMode: func.isRequired,
    removeCompleted: func.isRequired,
    changeSortMode: func.isRequired,
    currentMode: string.isRequired
};

export default FilterBar;