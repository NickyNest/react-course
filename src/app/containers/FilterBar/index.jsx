import React from 'react';
import {func, string} from 'prop-types';
import {Button} from 'semantic-ui-react';
import {connect} from 'react-redux';
import * as actions from 'utils/actions';
import {setShowMode, setSortMode} from '../../actions';

export const FilterBarPure = ({changeShowMode, removeCompleted, changeSortMode, currentMode}) => {
    const onClick = showMode => {
        changeShowMode(showMode);
    };

    const onChageSort = sortMode => {
        changeSortMode(sortMode);
    };

    return (
        <div>
            <Button id='btnShowAll' onClick={() => onClick(actions.SHOW_ALL)}>All</Button>
            <Button id='btnShowCompleted' onClick={() => onClick(actions.SHOW_COMPLETED)}>Completed</Button>
            <Button id='btnShowPending' onClick={() => onClick(actions.SHOW_PENDING)}>Pending</Button>
            <Button onClick={removeCompleted} disabled={currentMode === actions.SHOW_PENDING}>Remove completed</Button>
            <Button id='btnSortDateUp' onClick={() => onChageSort(actions.SORT_CREATED_DATE_UP)}>Date UP</Button>
            <Button id='btnSortDateDown' onClick={() => onChageSort(actions.SORT_CREATED_DATE_DOWN)}>Date DOWN</Button>
        </div>
    );
};

FilterBarPure.propTypes = {
    changeShowMode: func.isRequired,
    removeCompleted: func.isRequired,
    changeSortMode: func.isRequired,
    currentMode: string.isRequired
};

const mapStateToProps = state => ({
    currentMode: state.showMode
});

const mapDispatchToProps = dispatch => ({
    changeShowMode: showMode => dispatch(setShowMode(showMode)),
    removeCompleted: () => {},
    changeSortMode: sortMode => dispatch(setSortMode(sortMode))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(FilterBarPure);