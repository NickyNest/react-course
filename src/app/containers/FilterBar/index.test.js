/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilterBar from './index';

configure({adapter: new Adapter()});

describe('FilterBar', () => {
    it('with currentMode=SHOW_PENDING should match expected snapshot', () => {
        expect(shallow(<FilterBar
            changeShowMode={() => {}}
            removeCompleted={() => {}}
            changeSortMode={() => {}}
            currentMode='SHOW_PENDING' />)).toMatchSnapshot();
    });

    it('with currentMode=SHOW_COMPLETED should match expected snapshot', () => {
        expect(shallow(<FilterBar
            changeShowMode={() => {}}
            removeCompleted={() => {}}
            changeSortMode={() => {}}
            currentMode='SHOW_COMPLETED' />)).toMatchSnapshot();
    });

    it('with click on button ShowAll should call changeShowMode with parameter SHOW_ALL', () => {
        const changeShowModeMock = jest.fn();
        const wrapper = shallow(<FilterBar
            changeShowMode={changeShowModeMock}
            removeCompleted={() => {}}
            changeSortMode={() => {}}
            currentMode='SHOW_COMPLETED' />);
        wrapper.find('#btnShowAll').simulate('click');

        expect(changeShowModeMock.mock.calls.length).toEqual(1);
        expect(changeShowModeMock.mock.calls[0][0]).toEqual('SHOW_ALL');
    });

    it('with click on button ShowCompleted should call changeShowMode with parameter SHOW_COMPLETED', () => {
        const changeShowModeMock = jest.fn();
        const wrapper = shallow(<FilterBar
            changeShowMode={changeShowModeMock}
            removeCompleted={() => {}}
            changeSortMode={() => {}}
            currentMode='SHOW_COMPLETED' />);
        wrapper.find('#btnShowCompleted').simulate('click');

        expect(changeShowModeMock.mock.calls.length).toEqual(1);
        expect(changeShowModeMock.mock.calls[0][0]).toEqual('SHOW_COMPLETED');
    });

    it('with click on button ShowPending should call changeShowMode with parameter SHOW_PENDING', () => {
        const changeShowModeMock = jest.fn();
        const wrapper = shallow(<FilterBar
            changeShowMode={changeShowModeMock}
            removeCompleted={() => {}}
            changeSortMode={() => {}}
            currentMode='SHOW_COMPLETED' />);
        wrapper.find('#btnShowPending').simulate('click');

        expect(changeShowModeMock.mock.calls.length).toEqual(1);
        expect(changeShowModeMock.mock.calls[0][0]).toEqual('SHOW_PENDING');
    });

    it('with click on button RemoveCompleted should call removeCompleted once', () => {
        const removeCompletedMock = jest.fn();
        const wrapper = shallow(<FilterBar
            changeShowMode={() => {}}
            removeCompleted={removeCompletedMock}
            changeSortMode={() => {}}
            currentMode='SHOW_COMPLETED' />);
        wrapper.find('[disabled=false]').simulate('click');

        expect(removeCompletedMock.mock.calls.length).toEqual(1);
    });

    it('with click on button SortDateUp should call changeSortMode with parameter SORT_CREATED_DATE_UP', () => {
        const changeSortModeMock = jest.fn();
        const wrapper = shallow(<FilterBar
            changeShowMode={() => {}}
            removeCompleted={() => {}}
            changeSortMode={changeSortModeMock}
            currentMode='SHOW_COMPLETED' />);
        wrapper.find('#btnSortDateUp').simulate('click');

        expect(changeSortModeMock.mock.calls.length).toEqual(1);
        expect(changeSortModeMock.mock.calls[0][0]).toEqual('SORT_CREATED_DATE_UP');
    });

    it('with click on button SortDateDown should call changeSortMode with parameter SORT_CREATED_DATE_DOWN', () => {
        const changeSortModeMock = jest.fn();
        const wrapper = shallow(<FilterBar
            changeShowMode={() => {}}
            removeCompleted={() => {}}
            changeSortMode={changeSortModeMock}
            currentMode='SHOW_COMPLETED' />);
        wrapper.find('#btnSortDateDown').simulate('click');

        expect(changeSortModeMock.mock.calls.length).toEqual(1);
        expect(changeSortModeMock.mock.calls[0][0]).toEqual('SORT_CREATED_DATE_DOWN');
    });
});
