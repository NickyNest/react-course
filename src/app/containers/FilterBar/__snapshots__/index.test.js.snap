// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar with currentMode=SHOW_COMPLETED should match expected snapshot 1`] = `
<div>
  <Button
    as="button"
    id="btnShowAll"
    onClick={[Function]}
  >
    All
  </Button>
  <Button
    as="button"
    id="btnShowCompleted"
    onClick={[Function]}
  >
    Completed
  </Button>
  <Button
    as="button"
    id="btnShowPending"
    onClick={[Function]}
  >
    Pending
  </Button>
  <Button
    as="button"
    disabled={false}
    onClick={[Function]}
  >
    Remove completed
  </Button>
  <Button
    as="button"
    id="btnSortDateUp"
    onClick={[Function]}
  >
    Date UP
  </Button>
  <Button
    as="button"
    id="btnSortDateDown"
    onClick={[Function]}
  >
    Date DOWN
  </Button>
</div>
`;

exports[`FilterBar with currentMode=SHOW_PENDING should match expected snapshot 1`] = `
<div>
  <Button
    as="button"
    id="btnShowAll"
    onClick={[Function]}
  >
    All
  </Button>
  <Button
    as="button"
    id="btnShowCompleted"
    onClick={[Function]}
  >
    Completed
  </Button>
  <Button
    as="button"
    id="btnShowPending"
    onClick={[Function]}
  >
    Pending
  </Button>
  <Button
    as="button"
    disabled={true}
    onClick={[Function]}
  >
    Remove completed
  </Button>
  <Button
    as="button"
    id="btnSortDateUp"
    onClick={[Function]}
  >
    Date UP
  </Button>
  <Button
    as="button"
    id="btnSortDateDown"
    onClick={[Function]}
  >
    Date DOWN
  </Button>
</div>
`;
